### PR TITLE
Add HDF5 version 1.13.0.

### PR DESCRIPTION
**Pull request by lrknox**

### CODE REVIEW

**Comment by lrknox** [comment]
@chuckatkins I've added this PR for the new HDF5 1.13.0 "experimental" release.  It should pass the CI silo test, because it will build silo@4.2.10 with hdf5@1.10.7 by default.  HDF5 1.13.0 will be available, but hdf5@1.10.7 will be the preferred hdf5 version in spack until PR #27613 is approved.  Your approval of that one was cancelled when I added the conflicts statement that you suggested, so please review and approve both if at all possible.

**Review by brtnfld — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]
Confirmed sha256.  Will defer to others (e.g., @scheibelp ) for the merge.

**Comment by scheibelp** [comment]
I think it makes sense to re-run the pipeline here now that #27613 is merged (but I assume that will pass and this looks fine)

@spackbot re-run pipeline

**Pull request merged**
